src/lib.rs

```diff
@@ -35,3 +35,34 @@ pub use crate::config::{
 pub use crate::subscriber::{
     configure_subscriber, init_subscriber, ContextInfo, EventFormatter, Tracing, Uninstall,
 };
+
+/// Create a tracing error event, casting the error to &dyn [std::error::Error] for [layer::ErrorLayer],
+/// and adding the type name as error.kind.
+///
+/// Usage:
+/// ```
+/// use prima_tracing::report_error;
+///
+/// let error = "not a number".parse::<usize>().unwrap_err();
+/// report_error!(&error, "Parsing error!");
+/// ```
+///
+/// You can also add use add attributes and do things, just like with a regular [tracing::error]
+/// macro call
+/// ```
+/// use prima_tracing::report_error;
+/// # let input = "not a number";
+/// # let uid = "1223";
+///
+/// let error = input.parse::<usize>().unwrap_err();
+/// report_error!(&error, input, user=uid, "Parsing error: {}", &error);
```

Comment:
No reference needed if I'm not missing something:
```suggestion
/// report_error!(error, input, user=uid, "Parsing error: {}", error);
```

Reply:
Definitely needed. We could move the reference inside the macro instead but I think having it outside is cleaner

Reply:
Ah, I see now, it's because of the case. Well, I would argue that in Rust it's customary for these kind of macros to specify arguments without an explicit reference, but not take ownership of the arguments in any way. E.g. `std`'s `println!`, `format!` behave this way. And `tracing`'s macros also follow the same suit, e.g. `tracing::error!(error, "Error!"))` will not take ownership of `error`.

Reply:
really good point, agreed